File: CS_NET Web App/CS_NET Web App/Controllers/MemberController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using CS_NET_Web_App.Models;

namespace CS_NET_Web_App.Controllers
{
  public class MemberController : Controller
  {
    // GET: Member
    public ActionResult Index()
    {
      if (Session["memberId"] == null)
      {
        return RedirectToAction("Index", "Login");
      }

      var member = new MemberVM();
      var memberRewardsClient = new SBOfferSvc.OfferSoapClient("OfferSoap");
      SBOfferSvc.MemberActiveRewardsReturn memberRewardsReturn =
        memberRewardsClient.MemberActiveRewards(System.Configuration.ConfigurationManager.AppSettings["WSToken"],
        Session["memberId"].ToString(), -1, "PAW");
      if(memberRewardsReturn.ReturnCode != 0)
      {
        ViewBag.ErrorMessage = "There was an error getting your rewards";
      } else
      {
        member.Rewards = memberRewardsReturn.MemberActiveRewards.ToList();
      }
      return View(member);
    }
  }
}

File: VB_NET Web App/VB_NET Web App/Controllers/MemberController.vb
Imports System.Web.Mvc

Namespace Controllers
  Public Class MemberController
    Inherits Controller

    ' GET: Member/Details/5
    Function Index() As ActionResult
      If Session("memberId") Is Nothing Then
        Return RedirectToAction("Index", "Login")
      Else
        Dim member As New MemberVM
        Dim memberRewardsClient = New SBOfferSvc.OfferSoapClient("OfferSoap")
        Dim memberRewardsReturn As SBOfferSvc.MemberActiveRewardsReturn = memberRewardsClient.MemberActiveRewards(ConfigurationManager.AppSettings("WSToken"), Session("memberId"), -1, "PAW")
        If memberRewardsReturn.ReturnCode <> 0 Then
          ViewBag.ErrorMessage = "There was an error getting your rewards."
        Else
          member.Rewards = memberRewardsReturn.MemberActiveRewards.ToList()
        End If
        Return View(member)
      End If
    End Function

  End Class
End Namespace